File: solutions/leetcode/pairs-of-songs-with-total-durations-divisible-by-60/main.cpp
class Solution {
   public:
    int numPairsDivisibleBy60(vector<int>& time) {
        int result = 0;

        unordered_map<int, int> prefix = {};

        for (int t : time) {
            int v = t % 60;
            result += prefix[v];
            ++prefix[(60 - v) % 60];
        }

        return result;
    }
};

File: solutions/leetcode/pairs-of-songs-with-total-durations-divisible-by-60/main.py
class Solution:

    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        prefix = defaultdict(int)

        result = 0

        for t in time:
            v = t % 60
            result += prefix[v]
            prefix[(60 - v) % 60] += 1

        return result
